src/catchup/CatchupRange.cpp

```diff
@@ -37,4 +39,45 @@ operator!=(CatchupRange const& x, CatchupRange const& y)
 {
     return !(x == y);
 }
+
+CatchupRange
+makeCatchupRange(ResolvedCatchupConfiguration const& configuration,
+                 HistoryManager const& historyManager)
+{
+    auto toCheckpoint =
+        historyManager.nextCheckpointLedger(configuration.toLedger() + 1) - 1;
+
+    auto firstNotKnownCheckpoint = [&]() {
```

Comment:
why a lambda here?

Reply:
no good reason, can be replaced by simple ?: statement in last line